tenant_schemas/models.py

```diff
@@ -28,7 +30,7 @@ class TenantMixin(models.Model):
     """
 
     domain_url = models.CharField(max_length=128, unique=True)
-    schema_name = models.CharField(max_length=63, unique=True,
+    schema_name = models.CharField(max_length=63, unique=unique_public_schema(),
```

Comment:
How will this work with migrations? If the uniqueness constraint is dynamically determined, won't there be complaints by Django when you switch the setting value?

How about making `unique=False` and using the `clean` method to ensure uniqueness is as you wish?